### PR TITLE
Select CA for enrollment (Closes #1160)

### LINKED ISSUE #1160: Uses wrong CA for enrolling users via wallet interface
When creating a new identity using a gateway and enrollmentID and secret the extension selects the first certificate authority listed in the connection profile rather than the certificate authroity of the organization the connection profile refers too. This means that the enrollment fails as the extension attempts to register with another CA.

Steps to reproduce:
1. Have connection profile with multiple CA's with the first being not from the org listed at the top of the CP.
2. Create gateway using that connection profile
3. Create new wallet and add identity
4. Go through flow and get authorization failure as chosen CA knows nothing about the user you wish to register

### PR DESCRIPTION
**Pull request by toggle007**
Signed-off-by: Akshat Shah <akshatshah@akshats-mbp.hursley.uk.ibm.com>

### CODE REVIEW

File: client/src/commands/UserInputUtil.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class UserInputUtil {
     static readonly IMPORT_WALLET: string = 'Specify an existing file system wallet';
     static readonly WALLET_NEW_ID: string = 'Create a new wallet and add an identity';
     static readonly ADD_IDENTITY_METHOD: string = 'Choose a method for adding an identity';
-    static readonly ADD_CERT_KEY_OPTION: string = 'Provide certificate and private key files';
+    static readonly  ADD_CERT_KEY_OPTION: string = 'Provide certificate and private key files';
```
Comment (Jakeeyturner) [inline client/src/commands/UserInputUtil.ts]: Could you remove the space you've added here please!

File: client/src/commands/addWalletIdentityCommand.ts [hunk 2]
```diff
@@ -197,7 +193,33 @@ export async function addWalletIdentity(walletItem: WalletTreeItem | IFabricWall
...
+
+            if (gatewayRegistryEntry.connectionProfilePath.endsWith('.json')) {
+                connectionProfile = JSON.parse(connectionProfileFile);
+            } else {
+                // Assume its a yml/yaml file type
+                connectionProfile = yaml.safeLoad(connectionProfileFile);
+            }
+
+            // Get a list of CAs
+            const caKeys: any[] = Object.keys(connectionProfile.certificateAuthorities);
```
Comment (Jakeeyturner) [inline client/src/commands/addWalletIdentityCommand.ts]: Could you make the type `string[]` please

File: client/test/fabric/FabricCertificateAuthority.test.ts [hunk 3]
```diff
@@ -43,46 +40,13 @@ describe('FabricCertificateAuthority', () => {
     describe('enroll', () => {
 
         it('should return certificate and key from enrollment using JSON connection profile', async () => {
```
Comment (Jakeeyturner) [inline client/test/fabric/FabricCertificateAuthority.test.ts]: Should probably change the name of this test to be more accurate now that the code has changed.
E.g. `should return certificate and key from enrollment`

**Comment by Jakeeyturner** [comment]
Could you also add a test for `addWalletIdentityCommand`, when the file read is a `yml` file

File: client/src/commands/addWalletIdentityCommand.ts (hunk 2 above)
Comment (toggle007) [reply client/src/commands/addWalletIdentityCommand.ts]: Done

File: client/test/fabric/FabricCertificateAuthority.test.ts (hunk 3 above)
Comment (toggle007) [reply client/test/fabric/FabricCertificateAuthority.test.ts]: Done

**Pull request merged**